base/loggers/OwnPatternFormatter.cpp

```diff
@@ -7,13 +7,118 @@
 #include <Interpreters/InternalTextLogsQueue.h>
 #include <Common/CurrentThread.h>
 #include <base/terminalColors.h>
-
+#include "Loggers.h"
 
 OwnPatternFormatter::OwnPatternFormatter(bool color_)
     : Poco::PatternFormatter(""), color(color_)
 {
 }
 
+void OwnPatternFormatter::formatExtendedJSON(const DB::ExtendedLogMessage & msg_ext, std::string & text) const
+{
+    DB::WriteBufferFromString wb(text);
+    const Poco::Message & msg = msg_ext.base;
+    DB::writeChar('{', wb);
+    writeCString("\"", wb);
+    writeCString("datetime", wb); // key
+    writeCString("\"", wb);
+    writeCString(":", wb);
+    writeCString("\"", wb);
+    /// Change delimiters in date for compatibility with old logs.
+    DB::writeDateTimeText<'.', ':'>(msg_ext.time_seconds, wb);
+
+    DB::writeChar('.', wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 100000) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 10000) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 1000) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 100) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 10) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 1) % 10), wb);
+    writeCString("\"", wb);
+
+    writeCString(",", wb);
+
+    writeCString("\"", wb);
+    writeCString("thread_id", wb);
+    writeCString("\"", wb);
+    writeCString(":", wb);
+    writeCString("\"", wb);
+    writeCString(" [ ", wb);
+    if (color)
+        writeString(setColor(intHash64(msg_ext.thread_id)), wb);
+    DB::writeIntText(msg_ext.thread_id, wb);
+    if (color)
+        writeCString(resetColor(), wb);
+    writeCString(" ] ", wb);
+    writeCString("\"", wb);
+
+    /// We write query_id even in case when it is empty (no query context)
+    /// just to be convenient for various log parsers.
+    writeCString(",", wb);
+
+    writeCString("\"", wb);
+    writeCString("query_id", wb);
+    writeCString("\"", wb);
+    writeCString(":", wb);
+    writeCString("\"", wb);
+    writeCString("{", wb);
+
+    if (color)
+        writeString(setColor(std::hash<std::string>()(msg_ext.query_id)), wb);
+    DB::writeString(msg_ext.query_id, wb);
```

Comment:
This is wrong, because the string is not escaped.

Reply:
Agreed. Will change this.

Reply:
change submitted in updated PR.